### PR TITLE
New Gramplet: Historical Context (Gramps 5.2) Gramplet will show historical events in the gramplet panel.

### PR DESCRIPTION
**Pull request by kajmikkelsen**
New Gramplet: Historical Context (Gramps 5.2)
Gramplet will show historical events in the gramplet panel.

### CODE REVIEW

**Comment by Call-Me-Dave** [comment]
Great addon thank you :) 

As for the generic subject line of the pull request

> Initial commit for production the gramplet 

can I suggest you change it to:

> New Gramplet: Historical Context (Gramps 5.2)

or similar and add a proper description here?

**Comment by Call-Me-Dave** [comment]
I like this addon; as I was an old time TMG user and [created](http://www.tmgtips.com/timelin6.htm) a few personalized [TMG Timelines](http://www.tmgtips.com/timelines.htm)  but they used ( a .DBF file and a .DBT file and a DOC file!) and  8 data fields. But I expect I'll just convert to your format even if it looses information...

Maybe the instructions to create the file can be adapted to your addon ? ( Auto-Creation of (TMG)Timeline Files http://www.tmgtips.com/timelin6.htm )

File: HistContext/HistContext.gpr.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+         version = '0.2.10',
+         fname="HistContext.py",
+         height = 20,
+         detached_width = 510,
+         detached_height = 480,
+         expand = True,
+         gramplet = "HistContext",
+         gramplet_title=_("Historical Context"),
+         gramps_target_version="5.2",
+         help_url="https://github.com/kajmikkelsen/TimeLineGramplet",
```
Comment (Call-Me-Dave) [inline HistContext/HistContext.gpr.py:15]: change this to point to the new help page on the Gramps wiki for this addon say

help_url="https://gramps-project.org/wiki/index.php/Addon:HistoricalContext",

File: HistContext/HistContext.gpr.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+register(GRAMPLET,
+         id="HistContext", 
+         name=_("Historical Context"),
+         description = _("Displaying a historical context"),
+         status = BETA,
```
Comment (Call-Me-Dave) [inline HistContext/HistContext.gpr.py:5]: change status to "STABLE"

Comment (kajmikkelsen) [reply HistContext/HistContext.gpr.py:5]: Done

File: HistContext/README.md [hunk 3]
```diff
@@ -1,9 +1,14 @@
 ### Histtory Context gramplet
 
+![](./gramplet.png) 
 Gramplet to display historical events with the year they happened, and optionally the year  they ended.
+
 A double-click on a row will open your browser at the provided Link
 
-The gramplet works on the person and the relations view.
+The gramplet works on the person and the relations view and will us the bith and death of the active person.
```
Comment (Call-Me-Dave) [inline HistContext/README.md:8]: relations  = "Relationships Category view"
bith = birth

**Comment by kajmikkelsen** [comment]
Thanks

⁣Hent BlueMail til Android ​

Den 4. nov. 2024 13.25, fra 13.25, Call-Me-Dave ***@***.***> skrev:

File: HistContext/README.md (hunk 3 above)
Comment (kajmikkelsen) [reply HistContext/README.md]: Changed person to People, us to use and relations to Relationship Category
changed birt to birth
Committed and pushed

**Comment by GaryGriffin** [comment]
All looks good to me. Any other reason not to publish?

**Comment by Call-Me-Dave** [comment]
@kajmikkelsen  Congratulations on having your first addon published 🎂 Thank you

**Comment by kajmikkelsen** [comment]
Thanks
/Kaj 


Den 9. nov. 2024 23.22, fra 23.22, Call-Me-Dave ***@***.***> skrev:

**Comment by Call-Me-Dave** [comment]
https://gramps-project.org/wiki/index.php/Addon:Historical_Context

https://gramps.discourse.group/t/my-first-gramplet-for-gramps-5-2/6348/71

**Comment by PQYPLZXHGF** [comment]
Thanks for this very much appreciated 🥇 

@emyoulation Notice you updating the wiki for [this](https://gramps-project.org/wiki/index.php/Addon:Historical_Context#Event_Lists), can you add mention of what @Call-Me-Dave  [discusses above](https://github.com/gramps-project/addons-source/pull/615#issuecomment-2453121206) for tmg timeline files, maybe somebody can convert those to be use with this? 

Do any other genealogy programs have a similar feature?

**Comment by emyoulation** [comment]
Didn't have any luck with the links tonite from my phone. So I'll have to explore from a laptop tomorrow.

But knee-jerk reaction is reluctance to link to docs for a product 10 years gone. Or to borrow content from a commercial product within the copyright window.

**Comment by wroldwiedbwe** [comment]
@emyoulation [Page](http://www.tmgtips.com/timelines.htm) shows **Last revised: 11/16/2008 09:20:26**  and shows timelines provided by **Contributor**s with no restriction mentioned on usage.  ( The ones with green dots came with the program so not an issue )

**Comment by emyoulation** [comment]
> timelines provided by Contributors with no restriction mentioned on usage. 

No restriction but no license or permission either, right?

I think we need explicit permission ... or a compatible copyright licensing statement. And that means the contributors cannot be 'unknown' either.

**Comment by kajmikkelsen** [comment]
In a discussion with with @hartenthaler I mentioned the idea of using remarks in the file, in order to use different formats.
Herman is maintaining a module in webtrees for historical data.
The remarks could have keywords like this:
```
########################################################
## FORMAT: gramps
## VERSION: 1.0 
## AUTHOR: Kaj Mikkelsen
## CONTACT: dummy@mydomain.dk
## DESCRIPTION_G: Danish Kings
## DESCRIPTION_L: Danske Konger
########################################################

```

**Comment by kajmikkelsen** [comment]
And with no keywords, we just use the current format

**Pull request merged**